text/0000-gem-exec.md

```diff
@@ -0,0 +1,66 @@
+- Feature Name: Add `gem exec` command to install & run an executable from a gem
+- Start Date: 2023-01-24
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Add a `gem exec` subcommand to RubyGems that will handle installing (if necessary) and running an executable from a gem (on rubygems.org or the user's configured gem server), regardless of whether that gem is currently installed.
+
+# Motivation
+
+Inspired by [`npx` / `npm exec`](https://docs.npmjs.com/cli/v9/commands/npm-exec). This already exists as a separate gem called [gemx](https://github.com/rubygems/gemx), but is far less discoverable and dependable as a standalone gem than as a built-in RubyGems command.
+
+# Guide-level explanation
+
+The `gem exec` command can be thought of as a shortcut to running `gem install` and then the executable from the installed gem.
+
+For example, `gem exec rails new .` will run `rails new .` in the current directory, without having to manually run `gem install rails` only to run `rails new` once, and no other `rails` commands (because `rails new` generates a Gemfile).
+
+For a more complete example:
+
+```sh
+$ gem exec --gem cocoapods -r '> 1' -r '< 1.3' -v -- pod install --no-color --help
```

Comment:
It would be great to support also version identifier like you can do with `gem install`. `gem exec rails:7.1 new .`

Reply:
that makes sense too, I'll add support for it.